google/cloud/bigtable/internal/data_connection_impl_test.cc

```diff
@@ -1636,6 +1631,16 @@ TEST(DataConnectionTest, AsyncReadRowFailure) {
   EXPECT_THAT(resp, StatusIs(StatusCode::kPermissionDenied));
 }
 
+TEST(MakeDataConnection, DefaultsOptions) {
+  auto conn = bigtable::MakeDataConnection(
+      Options{}
+          .set<UnifiedCredentialsOption>(MakeInsecureCredentials())
+          .set<GrpcNumChannelsOption>(1));
+  auto options = conn->options();
+  EXPECT_EQ(options.get<AuthorityOption>(), "bigtable.googleapis.com");
+  EXPECT_EQ(options.get<GrpcNumChannelsOption>(), 1);
```

Comment:
We check that the unset `AuthorityOption` gets defaulted to the right value. Then I set `GrpcNumChannelOption` to 1 so that:
* we can verify that a user supplied option is not overridden by the defaults
* the minimum number of stubs are created